### PR TITLE
TEST/APPS/IO_DEMO: Add option specify differen port for each server

### PR DESCRIPTION
**Pull request by leibin2014**

### CODE REVIEW

File: test/apps/iodemo/io_demo.cc [hunk 1]
```diff
@@ -526,11 +526,23 @@ class DemoClient : public P2pDemoCommon {
...
+        char str[64];
+        const char* server_addr;
+        int   port_num;
+
         memset(&connect_addr, 0, sizeof(connect_addr));
         connect_addr.sin_family = AF_INET;
-        connect_addr.sin_port   = htons(opts().port_num);
+        if (strchr(server, ':') == NULL) {
+            port_num = opts().port_num;
+            server_addr = server;
```
Comment (evgeny-leksikov) [inline test/apps/iodemo/io_demo.cc:539]: basically LGTM, I would suggest just minor usability improvement:
does it mean that command line could be:
io_demo [options] **port0 port1**... ? I guess it assumes if servers are running on the same host
maybe to rework it as follow:
io_demo [options] **:port0 :port1**...
to make it clearer
IMHO this way the code could look simpler as well - no need second branch
@yosefe WDYT?

**Review by evgeny-leksikov — APPROVED** [review]

File: test/apps/iodemo/io_demo.cc (hunk 1 above)
Comment (evgeny-leksikov) [reply test/apps/iodemo/io_demo.cc:539]: ahh.. I see, it handles 
`io_demo [options] -p port addr1 addr2`
and
`io_demo [options] -p port0 addr0 addr1:port addr2:port2`

File: test/apps/iodemo/io_demo.cc [hunk 2]
```diff
@@ -526,11 +526,23 @@ class DemoClient : public P2pDemoCommon {
         return !_error_flag;
     }
 
-    UcxConnection* connect(const char* server_addr) {
+    UcxConnection* connect(const char* server) {
         struct sockaddr_in connect_addr;
+        char str[64];
+        const char* server_addr;
+        int   port_num;
```
Comment (dmitrygx) [inline test/apps/iodemo/io_demo.cc]: pls, remove extra whitespaces

File: test/apps/iodemo/io_demo.cc [hunk 3]
```diff
@@ -842,6 +854,7 @@ static int parse_args(int argc, char **argv, options_t* test_opts)
         case 'h':
         default:
             std::cout << "Usage: io_demo [options] [server_address]" << std::endl;
+            std::cout << "   Or: io_demo [options] [server_address0:port0] [server_address1:port1]..." << std::endl;
```
Comment (dmitrygx) [inline test/apps/iodemo/io_demo.cc]:
```
            std::cout << "Usage: io_demo [options] [server_address]" << std::endl;
            std::cout << "       or io_demo [options] [server_address0:port0] [server_address1:port1]..." << std::endl;
```

**Review by dmitrygx — APPROVED** [review]

File: test/apps/iodemo/io_demo.cc (hunk 1 above)
Comment (leibin2014) [reply test/apps/iodemo/io_demo.cc:539]: > ahh.. I see, it handles
> `io_demo [options] -p port addr1 addr2`
> and
> `io_demo [options] -p port0 addr0 addr1:port addr2:port2`

@evgeny-leksikov Yes, you are right.

File: test/apps/iodemo/io_demo.cc (hunk 2 above)
Comment (leibin2014) [reply test/apps/iodemo/io_demo.cc]: @dmitrygx Fixed.

File: test/apps/iodemo/io_demo.cc (hunk 3 above)
Comment (leibin2014) [reply test/apps/iodemo/io_demo.cc]: @dmitrygx Fixed. Thanks!

**Review by yosefe — APPROVED** [review]

**Pull request merged**